clients/src/test/java/org/apache/kafka/clients/admin/internals/AdminFetchMetricsManagerTest.java

```diff
@@ -0,0 +1,173 @@
+        MetricName nodeLatencyMax1 = metrics.metricName("request-latency-max", "group", genericTag(connectionId1));
+        registerNodeLatencyMetric(connectionId1, nodeLatencyAvg1, nodeLatencyMax1);
+        adminFetchMetricsManager.recordLatency(connectionId1, 105);
+        adminFetchMetricsManager.recordLatency(connectionId1, 108);
+
+        assertEquals(6.5, metricValue(nodeLatencyAvg0), EPSILON);
+        assertEquals(8, metricValue(nodeLatencyMax0), EPSILON);
+        assertEquals(106.5, metricValue(nodeLatencyAvg1), EPSILON);
+        assertEquals(108, metricValue(nodeLatencyMax1), EPSILON);
+
+        mockSleepTimeWindow();
+        adminFetchMetricsManager.recordLatency(connectionId0, 11);
+        adminFetchMetricsManager.recordLatency(connectionId1, 111);
+        assertEquals(8, metricValue(nodeLatencyAvg0), EPSILON);
+        assertEquals(11, metricValue(nodeLatencyMax0), EPSILON);
+        assertEquals(108, metricValue(nodeLatencyAvg1), EPSILON);
+        assertEquals(111, metricValue(nodeLatencyMax1), EPSILON);
+
+        mockSleepTimeWindow();
+        assertEquals(11, metricValue(nodeLatencyAvg0), EPSILON);
+        assertEquals(11, metricValue(nodeLatencyMax0), EPSILON);
+        assertEquals(111, metricValue(nodeLatencyAvg1), EPSILON);
+        assertEquals(111, metricValue(nodeLatencyMax1), EPSILON);
+
+        mockSleepTimeWindow();
+        assertTrue(Double.isNaN(metricValue(nodeLatencyAvg0)));
+        assertTrue(Double.isNaN(metricValue(nodeLatencyMax0)));
+        assertTrue(Double.isNaN(metricValue(nodeLatencyAvg1)));
+        assertTrue(Double.isNaN(metricValue(nodeLatencyMax1)));
+
+        adminFetchMetricsManager.recordLatency(connectionId0, 500);
+        adminFetchMetricsManager.recordLatency(connectionId0, 600);
+        mockSleepTimeWindow();
+        adminFetchMetricsManager.recordLatency(connectionId1, 800);
+        adminFetchMetricsManager.recordLatency(connectionId1, 900);
+        assertEquals(550, metricValue(nodeLatencyAvg0), EPSILON);
+        assertEquals(600, metricValue(nodeLatencyMax0), EPSILON);
+        assertEquals(850, metricValue(nodeLatencyAvg1), EPSILON);
+        assertEquals(900, metricValue(nodeLatencyMax1), EPSILON);
+
+        mockSleepTimeWindow();
+        assertTrue(Double.isNaN(metricValue(nodeLatencyAvg0)));
+        assertTrue(Double.isNaN(metricValue(nodeLatencyMax0)));
+        assertEquals(850, metricValue(nodeLatencyAvg1), EPSILON);
+        assertEquals(900, metricValue(nodeLatencyMax1), EPSILON);
+
+        mockSleepTimeWindow();
+        assertTrue(Double.isNaN(metricValue(nodeLatencyAvg1)));
+        assertTrue(Double.isNaN(metricValue(nodeLatencyMax1)));
+    }
+
+    private Map<String, String> genericTag(String connectionId) {
+        Map<String, String> tags = new LinkedHashMap<>();
+        tags.put("node-id", "node-" + connectionId);
+        return tags;
```

Comment:
We could leverage the `singletonMap`, since it only has a single entry here.
For exmaple:
```java
return Collections.singletonMap("node-id", "node-" + connectionId);
```